v2.0.0 rc

f1ana opened the pull request:
Use httpclientfactory
Add unit tests, separate integration tests
Make more friendly to dependency injection
Targets .NET Standard 2.0 and .NET 6

bdullemond reviewed (approved):
This is almost exactly what I've been doing locally today to get around adding a web proxy to the HttpClient (i.e. implementing my own WebInterface and using it in a ForwardGeocoder copy). 
This commit would allow me to just use the new 2.0 verion of this library instead.

bdullemond commented on src/Nominatim.API/Address/AddressSearcher.cs:
```diff
@@ -21,7 +24,8 @@ public class AddressSearcher {
         /// </summary>
         /// <param name="URL">URL to Nominatim service.  Defaults to OSM demo site.</param>
-        public AddressSearcher(string URL = null) {
+        public AddressSearcher(INominatimWebInterface nominatimWebInterface, string URL = null) {
```
Need to add the param comment

f1ana committed:
v2.0.0

Use httpclientfactory
Add unit tests, separate integration tests
Make more friendly to dependency injection
Targets .NET Standard 2.0 and .NET 6
